src/document/doc-utils.js

```diff
@@ -192,6 +192,10 @@ function getInnerParts(doc) {
 }
 
 function stripTrailingHardline(doc, withInnerParts = false) {
+  if (typeof doc === "string") {
+    return doc.replace(/(?:\r?\n)*$/, "");
+  }
```

Comment:
Breaking change for plugin developers

Reply:
I'll can this in `multiparser.js`, when that option is true

Reply:
@sosukesuzuki `doc.utils` is not public api? I remember mentioned when you add `normalizeParts`?

Reply:
`doc.utils` is public api.

Reply:
No? https://github.com/prettier/prettier/pull/8476#discussion_r434645944 , I can't find docs about it.

@evilebottnawi @thorn0